sale_order_amount_to_invoice/models/sale_order.py

```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    untaxed_amount_to_invoice = fields.Monetary(
+        compute="_compute_untaxed_amount_to_invoice", store=True
+    )
+
+    @api.depends("order_line.untaxed_amount_to_invoice")
+    def _compute_untaxed_amount_to_invoice(self):
+        for rec in self:
```

Comment:
As this field is used in tree views to doing the sum like this will have poor perf.
Please use `read_group` + `sum` aggregator. You can find plenty of examples in odoo core and OCA :wink: 
Reason to make it stored?

Reply:
Hi @simahawk thank you for the feedback! The reason to have this field stored is that we personally use it in the automated actions and reports. Regarding read_group + sum I would appreciate your pointing to some code because I'm getting your idea only partially.

Reply:
He wanted to say you can use postgres aggregate function passed in read_group() in order to let postgres do the job instead of summing in the loop.